src/prefect/server/events/pipeline.py

```diff
@@ -0,0 +1,38 @@
+from typing import List
+
+import pendulum
+
+from prefect.server.events.schemas.events import ReceivedEvent
+from prefect.server.events.services import event_persister
+from prefect.server.services import task_run_recorder
+from prefect.server.utilities.messaging.memory import MemoryMessage
+
+
+class EventsPipeline:
+    @staticmethod
+    def events_to_messages(events) -> List[MemoryMessage]:
+        messages = []
+        for event in events:
+            received_event = ReceivedEvent(
+                received=pendulum.now("UTC"), **event.model_dump()
+            )
+            message = MemoryMessage(
+                data=received_event.model_dump_json().encode(),
+                attributes={"id": str(event.id), "event": event.event},
+            )
+            messages.append(message)
+        return messages
+
+    async def process_messages(self, messages: List[MemoryMessage]):
```

Comment:
In the future when shutting down the ephemeral version we can invoke this method on everything sitting inside of the in-memory events queue